tests/store/artifact/test_databricks_sdk_models_artifact_repo.py

```diff
@@ -0,0 +1,186 @@
+    ]
+
+    observed_artifacts = repo.list_artifacts("root/subdir1")
+    assert observed_artifacts == [
+        FileInfo(is_dir=False, path="root/subdir1/file3", file_size=3),
+    ]
+
+    observed_artifacts = repo.list_artifacts("root/subdir2")
+    assert observed_artifacts == [
+        FileInfo(is_dir=False, path="root/subdir2/file4", file_size=4),
+        FileInfo(is_dir=False, path="root/subdir2/file5", file_size=5),
+    ]
+
+
+def test_upload_to_cloud(mock_databricks_workspace_client, tmp_path):
+    # write some content to a file at a local path
+    file_name = "a.txt"
+    file_content = b"file_content"
+    local_file_path = tmp_path.joinpath(file_name)
+    local_file_path.write_bytes(file_content)
+
+    # assert that databricks sdk file upload function is called to upload that content
+    # to expected_remote_file_path
+    repo = DatabricksSDKModelsArtifactRepository(TEST_MODEL_NAME, TEST_MODEL_VERSION)
+    repo._upload_to_cloud(None, local_file_path, file_name)
+
+    expected_remote_file_path = f"{TEST_MODEL_BASE_PATH}/a.txt"
+    mock_databricks_workspace_client.files.upload.assert_called_once_with(
+        expected_remote_file_path, mock.ANY, overwrite=mock.ANY
+    )
+
+
+def test_download_from_cloud(mock_databricks_workspace_client, tmp_path):
+    # write some content to a file at a local path
+    file_name = "a.txt"
+    local_file_path = tmp_path.joinpath(file_name)
+
+    # assert that databricks sdk file download function is called to download
+    # from expected_remote_file_path to local_file_path
+    mock_databricks_workspace_client.files.download.return_value = DownloadResponse(
+        contents=io.BytesIO(b"file_content")
+    )
+    repo = DatabricksSDKModelsArtifactRepository(TEST_MODEL_NAME, TEST_MODEL_VERSION)
+    repo._download_from_cloud(file_name, local_file_path)
+
+    expected_remote_file_path = f"{TEST_MODEL_BASE_PATH}/a.txt"
+    mock_databricks_workspace_client.files.download.assert_called_once_with(
+        expected_remote_file_path
+    )
+
+
+def test_log_artifact(mock_databricks_workspace_client, tmp_path):
+    # write some content to a file at a local path
+    file_name = "a.txt"
+    file_content = b"file_content"
+    local_file_path = tmp_path.joinpath(file_name)
+    local_file_path.write_bytes(file_content)
+
+    # assert that databricks sdk file upload function is called to upload that content
+    # to expected_remote_file_path
+    repo = DatabricksSDKModelsArtifactRepository(TEST_MODEL_NAME, TEST_MODEL_VERSION)
+    repo.log_artifact(local_file_path, file_name)
+
+    expected_remote_file_path = f"{TEST_MODEL_BASE_PATH}/a.txt"
+    mock_databricks_workspace_client.files.upload.assert_called_once_with(
+        expected_remote_file_path, mock.ANY, overwrite=mock.ANY
+    )
+
+
+def test_mlflow_use_databricks_sdk_model_artifacts_repo_for_uc(tmp_path):
+    MLFLOW_USE_DATABRICKS_SDK_MODEL_ARTIFACTS_REPO_FOR_UC.set(True)
```

Comment:
We should use `monkeypatch.setenv` when we need to set an environment variable for testing purposes.

Reply:
```suggestion
def test_mlflow_use_databricks_sdk_model_artifacts_repo_for_uc(tmp_path, monkeypatch):
    monkeypatch.setenv("MLFLOW_USE_DATABRICKS_SDK_MODEL_ARTIFACTS_REPO_FOR_UC", "true")
```